docs/architecture/adr-001-umee-module.md

```diff
@@ -34,54 +34,134 @@ synthetic ERC-20 meTokens.
-validators controlled by Umee governance.
-
-The multisig will be constructed such that all constituents are the validators
-in the current epoch. The threshold of the multisig will be the total number of
-validators in the epoch. This means all validators will have to sign and broadcast
-their part of the multisig. If any signature is missing, a cross delegation
-cannot be made. We do not envision a penalty for missing multisig transactions
-but this can be revised in the future in addition to an incentive mechanism.
-
-The multisig account must exist on native ATOM chain, which can already exist
-from a previous epoch or if a validator set changes in an epoch. When an account
-does not exist on the native ATOM chain, it will have to be created by Umee via
-IBC by sending some tokens to that account.
-
-A separate relayer process will not only be responsible for relaying IBC packets
-to and from the source chain and Umee, but it will also be responsible for
-listening for these delegation events, and broadcasting the constructed multisig
-transaction to the source chain.
-
-Any new transfers during E<sub>1</sub> will be processed in the same manner during
-E<sub>2</sub>.
+The `x/umee` module's `ModuleAccount` will be represented as an [Interchain Account](https://github.com/cosmos/ibc/tree/master/spec/app/ics-027-interchain-accounts). This account will exist and be managed on the Umee
+network while being registered on the destination ATOM chain as an `Interchain Account`.
+The `ModuleAccount` will have permission to delegate funds on behalf of Umee users
+via the `Interchain Account` ICS standard.
+
+At the beginning of each `BeginBlock`, the `x/umee` module will check if the
+`ModuleAccount` pool address has been registered on the destination ATOM chain.
+If not, in `BeginBlock` we will register the account via `RegisterIBCAccount`.
+
+```go
+func (keeper Keeper) RegisterIBCAccount(ctx sdk.Context, owner sdk.AccAddress, srcPort, srcChannel string) error {
+  salt := keeper.GetIncrementalSalt(ctx)
+  err := keeper.iaKeeper.TryRegisterIBCAccount(ctx, sourcePort, sourceChannel, []byte(salt))
+  if err != nil {
+    return err
+  }
+
+  keeper.PushAddressToRegistrationQueue(ctx, sourcePort, sourceChannel, owner)
+  ctx.EventManager().EmitEvent(sdk.NewEvent("register-interchain-account",sdk.NewAttribute("salt", salt)))
+
+  return nil
+}
+```
+
+Note, we push the account into a queue via, `PushAddressToRegistrationQueue`, so
+that we can implement a hook where on packet ACK, the `x/umee` module can store
+and mark the account as registered<sup>1</sup>.
+
+Once the `ModuleAccount` pool address is registered, we are now able to perform
+cross-chain delegations on the destination ATOM chain. During each `EndBlock`
+execution in the `x/umee` module, we will construct a series of messages that will
+be executed on the source ATOM chain via `TryRunTx` as defined by the
+`Interchain Account` ICS standard. Finally, the `PendingDelegationQueue` will be
```

Comment:
I read through the interchain account standard and I didn't really see any mention of signers... 

all I see is a confusing paragraph on it from the spec

> It does not restrict how you can distinguish signers that was not made by the counterparty chain. But the most common way would be to record the account in state when the interchain account is registered and to verify that signers are recorded interchain account.

So does this mean that we still need to implement something that's effectively a weighted multisig? Is that what `ModuleAccount` is?

Reply:
Good question. **No, there no need for any multi-sig**. It's completely unnecessary. Validators are not responsible for signing anything in this context. Keep in mind, the multi-sig approach was always a temporary stop-gap approach until Interchain accounts became available.

Interchain accounts take care of authorization and authentication of cross-chain accounts and the messages they can execute. They do this through the Interchain account module, leveraging IBC's ports and channels.

That being said, a relayer is still responsible for signing and relaying the actual transactions. This is no different than any other IBC-type transaction.